internal/indexer/processors/contract_deploy_test.go

```diff
@@ -0,0 +1,259 @@
+package processors
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stellar/go/network"
+	operation_processor "github.com/stellar/go/processors/operation"
+	"github.com/stellar/go/xdr"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stellar/wallet-backend/internal/indexer/types"
+	"github.com/stellar/wallet-backend/internal/utils"
+)
+
+func Test_ContractDeployProcessor_Process_invalidOpType(t *testing.T) {
+	ctx := context.Background()
+	proc := NewContractDeployProcessor(network.TestNetworkPassphrase)
+
+	op := &operation_processor.TransactionOperationWrapper{
+		Operation: xdr.Operation{Body: xdr.OperationBody{Type: xdr.OperationTypePayment}},
+	}
+	changes, err := proc.ProcessOperation(ctx, op)
+	assert.ErrorIs(t, err, ErrInvalidOpType)
+	assert.Nil(t, changes)
+}
+
+func Test_ContractDeployProcessor_Process_createContract(t *testing.T) {
+	const (
+		opSourceAccount   = "GBZURSTQQRSU3XB66CHJ3SH2ZWLG663V5SWM6HF3FL72BOMYHDT4QTUF"
+		fromSourceAccount = "GCQIH6MRLCJREVE76LVTKKEZXRIT6KSX7KU65HPDDBYFKFYHIYSJE57R"
+		authSignerAccount = "GDG2KKXC62BINMUZNBTLG235323N6BOIR33JBF4ELTOUKUG5BDE6HJZT"
+	)
+
+	ctx := context.Background()
+
+	builder := NewStateChangeBuilder(12345, closeTime.Unix(), "0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20").
+		WithOperationID(53021371269121).
+		WithReason(types.StateChangeReasonDeploy).
+		WithCategory(types.StateChangeCategoryContract)
+
+	type TestCase struct {
+		name             string
+		op               *operation_processor.TransactionOperationWrapper
+		wantStateChanges []types.StateChange
+	}
+	testCases := []TestCase{}
+
+	for _, withSubinvocations := range []bool{false, true} {
```

Comment:
Really good comprehensive tests 👏 . Would you mind adding some comments here for someone who is not well versed with the XDR objects and contract ops. That would be helpful